WorkoutTest/ProgramViewController.swift

```diff
@@ -98,11 +119,36 @@ extension ProgramViewController {
   }
 }
 
+//MARK: - UISearchBarDelegate
+
+extension ProgramViewController: UISearchBarDelegate{
+  func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+    filterPrograms(with: searchText)
+  }
+  
+  func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+    searchBar.text = ""
+    searchBar.resignFirstResponder()
+    applySnapshot()
+  }
+  
+  private func filterPrograms(with searchText: String) {
+    let filteredPrograms = welcome?.programs.filter({
+      searchText.isEmpty || $0.name.lowercased().contains(searchText.lowercased())
+    }) ?? []
+    
+    var snapShot = NSDiffableDataSourceSnapshot<ProgramListSection,Program>()
+    snapShot.appendSections([.programs])
+    snapShot.appendItems(filteredPrograms, toSection:.programs)
+    dataSource.apply(snapShot, animatingDifferences: true)
```

Comment:
what would happen if you type something into the search bar, and then make 'pull to refresh' ?